### DISCUSSION TITLE
Retaining Original Values for Specific Identifiers during Anonymization in Presidio Structured

### DISCUSSION THREAD

**Discussion by ardband** [microsoft/presidio#1318] [Q&A]
I want to retain the original values for certain identifiers like email addresses, but the current output shows <None> after anonymization. Is there a way to achieve this without applying any anonymization logic to specific identifiers?

code-
```py
sample_df = pd.DataFrame({'name': ['John Doe', 'Jane Smith'], 'email': ['john.doe@example.com', 'jane.smith@example.com']})

tabular_analysis = PandasAnalysisBuilder().generate_analysis(sample_df)

fake = Faker()
operators = {
  "PERSON": OperatorConfig("replace", {"new_value": "REDACTED"}),
  "EMAIL_ADDRESS": OperatorConfig("custom", {"lambda": lambda x: fake.safe_email()})
}

anonymized_df = pandas_engine.anonymize(sample_df, tabular_analysis, operators=operators)
print(anonymized_df)
```
output-
       name   email
0  REDACTED  <None>
1  REDACTED  <None>

**Comment by omri374** [comment]
It should be possible to integrate something like this [pseudonymization sample](https://github.com/microsoft/presidio/blob/main/docs/samples/python/pseudonomyzation.ipynb) into the presidio-structured flow. 
It has similar functionality for what you're describing, and also allows you to de-anonymize later. The main difference would that in the sample, the sentence "My name is Rick" would be replaced with "My name is <PERSON_0> and with presidio-structured, the sentence "My name is Rick" would be replaced with "<PERSON_0>". I haven't tested this myself though.
